### PR TITLE
docs: Extend design doc for HIP-719

### PR DESCRIPTION
**Pull request by mustafauzunn**
**Description**:

Extend design doc for HIP-719

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by lukelee-sl — APPROVED** [review]
LGTM.  tyvm!

File: hedera-node/docs/design/services/smart-contract-service/IHRC-associate-dissociate-isassociated.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+## Non Goals
+
+- Refactor the current implementation to better support future additions to the token facade contract functionality.
+
+## Architecture
+
+The method in which the redirection to proxy contract is implemented is that during the execution of an EVM request, we examine the contract address to see
+if it is an HTS token address and if so we redirect the request to the proxy contract.  
+
+More specifically the algorithm by which we get from (a) the call to the token address to (b) the precompiled contract is a follows:
```
Comment (MrValioBg) [inline hedera-node/docs/design/services/smart-contract-service/IHRC-associate-dissociate-isassociated.md:23]:
```suggestion
More specifically the algorithm by which we get from (a) the call to the token address to (b) the precompiled contract is as follows:
```

File: hedera-node/docs/design/services/smart-contract-service/IHRC-associate-dissociate-isassociated.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+The `IsAssociatedCall` will override `resultOfViewingToken` method and return the result of the `isAssociated` function.
+
+## Acceptance Tests
+
+### Positive Tests
+- Create a contract that performs the `associate`, `dissociate` and `isAssociated` functions on a fungible token and ensure that the functions can be called successfully by an EOA.
+- Create a contract that performs the `associate`, `dissociate` and `isAssociated` functions on an NFT and ensure that the functions can be called successfully by an EOA.
+- Create a contract that performs the `associate`, `dissociate` and `isAssociated` functions on a fungible token and ensure that the functions can be called successfully by a contract.
+- Create a contract that performs the `associate`, `dissociate` and `isAssociated` functions on an NFT and ensure that the functions can be called successfully by a contract.
+- Ensure that the `associate` and `dissociate` functions disregards the signature when called via the token facade.
```
Comment (MrValioBg) [inline hedera-node/docs/design/services/smart-contract-service/IHRC-associate-dissociate-isassociated.md:98]:
```suggestion
- Ensure that the `associate` and `dissociate` functions disregard the signature when called via the token facade.
```

File: hedera-node/docs/design/services/smart-contract-service/IHRC-associate-dissociate-isassociated.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+			let size := returndatasize()
+			returndatacopy(0, 0, size)
+			switch result
+				case 0 { revert(0, size) }
+				default { return(0, size) }
+		}
+	}
+}
+```
+5. This means that _any_ function can be redirected-to as long as the HTS precompile handles the redirect call [here](https://github.com/hashgraph/hedera-services/blob/a1ccc19042d577c84076e97ee8485f33e2c9e696/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/processors/CustomMessageCallProcessor.java#L121).
```
Comment (MrValioBg) [inline hedera-node/docs/design/services/smart-contract-service/IHRC-associate-dissociate-isassociated.md:48]:
```suggestion
5. This means that _any_ function can be redirected to as long as the HTS precompile handles the redirect call [here](https://github.com/hashgraph/hedera-services/blob/a1ccc19042d577c84076e97ee8485f33e2c9e696/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/processors/CustomMessageCallProcessor.java#L121). 
```

**Review by stoyanov-st — DISMISSED** [review]
tyvm! LGTM.

**Review by MrValioBg — DISMISSED** [review]
LGTM! I left comments for a few minor corrections.

**Review by stoyanov-st — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @MrValioBg !

**Pull request merged**